### PR TITLE
Upgrade to version 0.25.0

### PR DESCRIPTION
**Pull request by pipo02mix**
New version [0.25.0](https://github.com/kubernetes/ingress-nginx/blob/master/Changelog.md) has been released with validation controller included.

### CODE REVIEW

**Comment by pipo02mix** [comment]
I have omitted the webhook controller for now because it is was not working well. But I have taken the chance to remove the default http backend, no needed anymore, plus the integration test for this and I have added the changes in RBAC

**Comment by pipo02mix** [comment]
tested on ginger

File: integration/templates/ingress_controller_basic_values.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ controller:
 
   image:
     repository: giantswarm/nginx-ingress-controller
-    tag: 0.23.0
```
Comment (tomahawk28) [inline integration/templates/ingress_controller_basic_values.go]: it was not `0.24.1` before, make senses to change atm?

File: integration/templates/ingress_controller_migration_values.go [hunk 2]
```diff
@@ -17,7 +17,7 @@ controller:
 
   image:
     repository: giantswarm/nginx-ingress-controller
-    tag: 0.23.0
```
Comment (tomahawk28) [inline integration/templates/ingress_controller_migration_values.go]: This as well.

**Review by tomahawk28 — APPROVED** [review]

**Comment by jgsqware** [comment]
I think it would be great to have a summary of good point of the upgrade and attention point too to give to customer

File: integration/templates/ingress_controller_basic_values.go (hunk 1 above)
Comment (pipo02mix) [reply integration/templates/ingress_controller_basic_values.go]: It is only informational, but I would like to have it aligned. @rossf7 WDYT?

**Comment by pipo02mix** [comment]
Good point, I need to add the CHANGELOG entry

**Review by jgsqware — APPROVED** [review]
LGTM

File: integration/templates/ingress_controller_basic_values.go (hunk 1 above)
Comment (rossf7) [reply integration/templates/ingress_controller_basic_values.go]: Yes this is just for the e2e test and I don't like that we need it. But yes it should be aligned I guess this just got missed last time.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,13 +5,24 @@ All notable changes to this project will be documented in this file.
...
+
+### Added
+
+- Upgrade nginx controller to use the latest version [0.25.0](https://github.com/kubernetes/ingress-nginx/releases/tag/nginx-0.25.0). Notable changes there:
+
+    - Support new [networking.k8s.io/v1beta1 package](https://github.com/kubernetes/ingress-nginx/pull/4127).
+
+    - Migration from Nginx to OpenResty 1.15.8.
+
+    - This version add a validation webhook controller for ingress resource validation but it has not been enabled yet as we found some issues during the testing process.
```
Comment (rossf7) [inline CHANGELOG.md]:
```suggestion
    - This version adds a validation webhook controller for ingress resource validation but it has not been enabled yet. As we found some issues during the testing process.
```

**Review by rossf7 — APPROVED** [review]
LGTM

Nice that we simplify and can remove the default backend. ❤️

**Pull request merged**